chain/types/ethtypes/errors.go

```diff
@@ -0,0 +1,43 @@
+package ethtypes
+
+import (
+	"fmt"
+
+	"golang.org/x/xerrors"
+)
+
+var ErrExecutionReverted = xerrors.New("execution reverted")
+
+// ExecutionRevertedError is an error that occurs when a transaction reverts.
+type ExecutionRevertedError struct {
+	Message string
+	Code    int
+	Data    string
+}
+
+// Error implements the error interface.
+func (e *ExecutionRevertedError) Error() string {
+	if e.Message == "" {
+		return fmt.Sprintf("json-rpc error %d", e.Code)
+	}
```

Comment:
do we expect to ever hit this?

maybe we should do away with `Message` entirely and just always return this, make it `("execution reverted: %d", e.Code)`

Reply:
and if we don't hit this code, I wonder what the point of `Code` is, do we use that anywhere?

Reply:
Yeah so If you look at the `ErrorCode()` function it returns the code, which is basically for not registered errors.

If you look at this comment above I think you will get the idea: https://github.com/filecoin-project/lotus/pull/12553#issuecomment-2414106748
Whatever we decide based on it, we will either remove the code or keep it.